#!/bin/bash

#SBATCH -o /home/mayran/error/slurm-%A_%2a.out
#SBATCH -e /home/mayran/error/slurm-%A_%2a.err
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=alexandre.mayran@epfl.ch
#SBATCH --nodes 1
#SBATCH --mem 120G
#SBATCH --cpus-per-task 1
#SBATCH --time 8:00:00
#SBATCH --array=2-7
#SBATCH --job-name Cell.ranger
#SBATCH --chdir /scratch/mayran/scRNA/

pathWithGitHub=$1

pathForTableWithSamples=${pathWithGitHub}/scRNAseq/metadata.txt
sample=$(cat $pathForTableWithSamples | awk -v i=$SLURM_ARRAY_TASK_ID 'NR==i{print $1}')

# The transcrptome has been previously generated by
# https://github.com/lldelisle/scriptsForBoltEtAl2021/blob/main/scRNAseq/0_prepare_cellrangerv3.1.0.sh
# Using version 3.1.0, fasta from UCSC mm10 and gtf from
# "https://zenodo.org/record/4456702/files/mergeOverlapGenesOfFilteredTranscriptsOfMus_musculus.GRCm38.98_ExonsOnly_UCSC.gtf.gz?download=1"

# Here cellranger version 6.0.0 is used:
cellranger count --id=$sample \
    --transcriptome=/home/mayran/genome/merged.filtered.ensembl.grc38.98 \
    --localcores=8 \
    --fastqs=fastqs/$sample
